src/main/java/io/cdap/plugin/gcp/spanner/source/SpannerSourceConfig.java

```diff
@@ -199,12 +204,23 @@ public boolean canConnect() {
   }
 
   /**
-   * Get fully-qualified name (FQN) for a Spanner table (FQN format: spanner://{instanceId}.{database}.{table}).
+   * Get fully-qualified name (FQN) for a Spanner table
+   * (FQN format: spanner:{projectId}.{instanceConfigId}.{instanceId}.{databaseId}.{tableId}).
    *
    * @return String fqn
    */
-  public String getFQN() {
-    String secondFQNPart = String.join(".", instance, database, table);
-    return SpannerConstants.SPANNER_FQN_PREFIX + secondFQNPart;
+  public String getFQN(String instanceConfigId) {
+    String formattedProjectId = SpannerUtil.formatAsFQNComponent(connection.getProject());
+    String formattedInstanceConfigId = SpannerUtil.formatAsFQNComponent(instanceConfigId);
+    String formattedInstance = SpannerUtil.formatAsFQNComponent(instance);
+    String formattedDatabase = SpannerUtil.formatAsFQNComponent(database);
+    String formattedTable = SpannerUtil.formatAsFQNComponent(table);
+
+    String fqn = String.format("%s:%s.%s.%s.%s.%s", SpannerConstants.SPANNER_FQN_PREFIX,
+        formattedProjectId, formattedInstanceConfigId, formattedInstance,
+        formattedDatabase, formattedTable);
+
+    LOG.trace("Formatted Fully-Qualified Name (FQN): {}", fqn);
+    return fqn;
```

Comment:
Looks like code getting duplicated here in both `SourceConfig` and `SinkConfig` class, can you plz move this method to `SpannerUtil`?

Reply:
Moved the `getFQN` function to `SpannerUtil`